lib/govuk_ab_testing/experiment.rb

```diff
@@ -0,0 +1,36 @@
+module GovukAbTesting
+  class Experiment
+    attr_reader :ab_test, :request
+
+    delegate :request_header, :cookie_name, :response_header, to: :ab_test
+
+    # @param experiment_name [String] Lowercase experiment name, like `example`
+    # @param request [ApplicationController::Request] the `request` in the
+    # controller.
+    def initialize(experiment_name, request)
+      @ab_test = AbTest.new(experiment_name)
+      @request = request
+    end
+
+    # Get the bucket this user is in
+    #
+    # @return [String] the current variant, "A" or "B"
+    def variant_name
+      request.headers[request_header] == "B" ? "B" : "A"
+    end
+
+    # Configure the response
+    #
+    # @param [ApplicationController::Response] the `response` in the controller
+    def add_response_header(response)
+      response.headers['Vary'] = response_header
```

Comment:
Do we have to worry about this overriding exiting `Vary` headers?

I used this in `HelpController` in frontend because I was sure that it wasn't overriding any extra `Vary` headers, but is it safe to make that assumption in every app?